### DISCUSSION TITLE
Add number to toggle-wrapper span class based on the number of rows in pivot table

### DISCUSSION THREAD

**Discussion by Wojtek17** [apache/superset#24764] [Ideas]
Hi! I have a complicated issue. I have 4 levels of rows in pivot table. I need for each deeper level bigger margin. The problem is that I can't do that using css and the nth child method as the table is dynamic. There is a level for "B.II". Let's say it's 11nth child of the tbody. The problem is that if I now roll up the A.I level, the B.II it won't be 11nth child anymore right? 

![Zrzut ekranu 2023-07-21 091242](https://github.com/apache/superset/assets/112469140/920e1810-85f8-459f-b391-dd3da48dee4d)

I found that the span class is defined in the react pivottable. I paste the full path: https://github.com/apache/superset/blob/master/superset-frontend/plugins/plugin-chart-pivot-table/src/react-pivottable/TableRenderers.jsx
 
The definiton of the class is at the beginning of the code.
  const name = namesMapping[value] || value;
  return needToggle ? (
span className="toggle-wrapper"
 

The question is: Is there any way to introduce the number to this span toggle-wrapper class that would be based on number of the level. So for each level 1 it would be toggle-wrapper-1 etc.

**Comment by michael-s-molina** [comment]
Pinging @kgabryje here as the author of Pivot Table v2. He's the best person to point you in the right direction for your problem.

**Comment by kgabryje** [comment]
Hey @Wojtek17, not sure if I understood correctly, but do you want to achieve something like this?

<img width="1726" alt="image" src="https://github.com/apache/superset/assets/15073128/7a710c8f-a8c1-4fe7-9dd9-c3c9454cadf8">

I think the easiest way to achieve it is to tie the padding value to level's index (the `&&&` is just to to increase the specificity):

<img width="585" alt="image" src="https://github.com/apache/superset/assets/15073128/b85a9c27-6eb3-45cf-92bf-f3f7ce6af5b4">

And similar if you want the same for rows

**Comment by Wojtek17** [comment]
Hey @kgabryje, thanks for the answer. Yes this is the result I want. Is it possible to do this for all levels except the first one? In this specific example the gender shouldn't have the margin/padding.

**Reply by kgabryje** [reply]
You can drop the `+ 1` I guess

**Reply by kgabryje** [reply]
Alternatively, change className to sth like ``className={`pvtAxisLabel pvtAxisLabel-${attrIdx}`}`` and then target `pvtAxisLabel-0`, `pvtAxisLabel-1` and so on in css

**Comment by Wojtek17** [comment]
Hey @kgabryje, the soultion you suggested doesn't compile. We are getting error that css variable doesn't exist. We are on 2.1.0 version
![image](https://github.com/apache/superset/assets/112469140/5348f2a7-f49f-45aa-820a-55ab5de97f77)

**Reply by kgabryje** [reply]
You can import css from `@superset-ui/core`

**Comment by Wojtek17** [comment]
Hey @kgabryje, thanks for the answers. We actually managed to resolve this challenge by adding additional id class in the spans as I described it in the beginning.

**Comment by michael-s-molina** [comment]
Nice! Thanks @kgabryje!
